### PR TITLE
✨ 95 build a wrapper component for icons

### PR DESCRIPTION
**Pull request by langehm**
Issues #95

### CODE REVIEW

**Comment by langehm** [comment]
Should we change out all of the icons in the other components in this issue/pr or open another?

File: src/components/Icon/MucIcon.stories.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  title: "MucIcon",
+  tags: ["autodocs"],
+  // 👇 Use `fn` to spy on the onClick arg, which will appear in the actions panel once invoked
+  args: { onClick: fn() },
+  parameters: {
+    docs: {
+      description: {
+        component: `The \`muc-icon\` component is a wrapper commponent for all icons in the patternlab.
+
+[🔗 Patternlab-Docs](https://patternlab.muenchen.space/?p=viewall-guidelines-icons)
```
Comment (FabianWilms) [inline src/components/Icon/MucIcon.stories.ts:16]:
```suggestion
[🔗 Patternlab-Docs](https://patternlab.muenchen.space/?p=viewall-guidelines-icons)
[🖼 Iconography](https://it-at-m.github.io/muc-patternlab-vue/?path=/docs/iconography--docs)
```

File: src/components/Icon/MucIcon.vue [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<template>
+  <svg
+    aria-hidden="true"
+    class="m-button__icon"
+    :style="{ color: color }"
+  >
+    <use :xlink:href="'#icon-' + icon" />
```
Comment (FabianWilms) [inline src/components/Icon/MucIcon.vue:7]:
```suggestion
    <use :href="'#icon-' + icon" />
```

xlink-href ist [deprecated](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/xlink:href?retiredLocale=de)

File: src/components/Icon/MucIcon.vue [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+<script setup lang="ts">
+defineProps<{
+  /**
+   * String of the icon to be displayed.
+   */
+  icon: string;
+
+  /**
+   * Optional color string for the icon.
```
Comment (FabianWilms) [inline src/components/Icon/MucIcon.vue:19]:
```suggestion
   * Optional css-color for the icon.
```

With your current Impl. I could also use `#121212` or `var(--my-color)`

**Review by FabianWilms — APPROVED** [review]

**Comment by FabianWilms** [comment]
> Should we change out all of the icons in the other components in this issue/pr or open another?

https://github.com/it-at-m/muc-patternlab-vue/issues/104

**Review by FabianWilms — APPROVED** [review]

**Pull request merged**